pkg/ipam/ipam_test.go

```diff
@@ -195,7 +198,7 @@ func TestIPAMMetadata(t *testing.T) {
 	require.Equal(t, PoolDefault(), resIPv4.IPPoolName)
```

Comment:
should we get rid of this assertions (and the corresponding fallback to the DefaultPool if the metadata manager isn't set? IMO - with the fallback in the metadata manager cell, this is actually no longer necessary.

Reply:
agree. will remove this and ensure that we don't lose test coverage

Reply:
i removed the fallback and updated the tests accordingly.